Justfile

```diff
@@ -0,0 +1,30 @@
+bats_invocation := "flox run '.#flox-tests' -- -- --flox target/debug/flox"
+cargo_test_invocation := "cargo test --workspace"
+
+_default:
+    @just --list --unsorted
+
+# Run the 'bats' test suite
+bats-tests +test_files="":
+    @cargo build
+    @{{bats_invocation}} {{test_files}}
+
+# Run the Rust unit tests
+unit-tests regex="":
+    @{{cargo_test_invocation}} {{regex}}
+
+# Run the test suite, including impure tests
+impure-tests regex="":
+    @{{cargo_test_invocation}} {{regex}} --features "extra-tests"
+
+# Run the entire test suite, not including impure tests
+test: unit-tests bats-tests
+
+# Run the entire test suite, including impure tests
+test-all: impure-tests bats-tests
+
+# Enters the Rust development environment
+work:
+    @# Note that this command is only really useful if you have
+    @# `just` installed outside of the `flox` environment already
```

Comment:
Yeah I personally won't use this one - I'd just run the flox command or put it in a .envrc